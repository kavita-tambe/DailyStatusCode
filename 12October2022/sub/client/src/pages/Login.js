import React, { useState } from "react";

import lock from "./lock1.jpg";
import "./Register.css";
import { Link } from "react-router-dom";
const Register = () => {
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();

  return (
    <>
      <div className="d-flex justify-content-center" style={{ height: "80vh" }}>
        <div className="wrapper">
          <div className="logo">
            <img src={lock} alt="" />
          </div>
          <div className="text-center mt-4 name">Login</div>
          <form className="p-3 mt-3">
            <div className="form-field d-flex align-items-center">
              <span className="fas fa-key"></span>
              <input
                type="text"
                value={email}
                // onClick={setEmail}
                placeholder="Email"
              />
            </div>
            <div className="form-field d-flex align-items-center">
              <span className="fas fa-key"></span>
              <input
                type="password"
                value={password}
                // onClick={setPassword}
                placeholder="Password"
              />
            </div>
            <button className="btn mt-3">Login</button>
          </form>
          <div className="text-center fs-6">
            <Link to="/register" variant="body2">
              Don't have an account? Sign Up here..
            </Link>
          </div>
        </div>
      </div>
    </>
  );
};

export default Register;
