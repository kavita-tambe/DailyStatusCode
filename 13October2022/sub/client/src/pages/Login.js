import React, { useState } from "react";

import lock from "./lock1.jpg";
import "./Register.css";
import { Link, useNavigate } from "react-router-dom";
const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");
  const [error, setError] = useState("");

  // const navigate = useNavigate();

  const emailChangeHandler = (e) => {
    setEmail(e.target.value);
  };
  const passwordChangeHandler = (e) => {
    setPassword(e.target.value);
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    const regexEmail = /^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,15}$/;
    if (!email || regexEmail.test(email) === false) {
      setEmailError("Email is not valid");
    } else {
      setEmailError("");
    }
    setError("");
    const regexPassword = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/i;

    if (!password || regexPassword.test(password) === false) {
      setPasswordError("Please enter the valid password");
    } else {
      setPasswordError("");
    }
    try {
      // navigate("/");
    } catch (err) {
      setError("Invalid email and password");
    }
  };
  const handleBlurEmail = (e) => {
    if (e.target.value) {
      setEmailError("");
      return true;
    } else {
      setEmailError("Email is required");
    }
  };
  const handleBlurPassword = (e) => {
    if (e.target.value) {
      setPasswordError("");
      return true;
    } else {
      setPasswordError("Password is required");
    }
  };
  const handleFocusEmail = () => {
    setEmailError("");
  };
  const handleFocusPassword = () => {
    setPasswordError("");
  };

  return (
    <>
      <div className="d-flex justify-content-center" style={{ height: "80vh" }}>
        <div className="wrapper">
          <div className="logo">
            <img src={lock} alt="" />
          </div>
          <div className="text-center mt-4 name">Login</div>
          <form className="p-3 mt-3" onSubmit={handleSubmit}>
            <div className="data-div">
              <div className="form-field d-flex align-items-center">
                <span className="fas fa-key"></span>
                <input
                  type="email"
                  value={email}
                  placeholder="Email"
                  name="email"
                  onChange={emailChangeHandler}
                  onBlur={handleBlurEmail}
                  onFocus={handleFocusEmail}
                />
              </div>
              <div>
                <span className="form-pass">{emailError}</span>
              </div>
            </div>
            <div className="data-div">
              <div className="form-field d-flex align-items-center">
                <span className="fas fa-key"></span>
                <input
                  type="password"
                  value={password}
                  placeholder="Password"
                  onChange={passwordChangeHandler}
                  onBlur={handleBlurPassword}
                  onFocus={handleFocusPassword}
                />
              </div>

              <div>
                <span className="form-pass">{passwordError}</span>
              </div>
            </div>

            <button className="btn login-text">Login</button>
          </form>
          <div className="text-center fs-6">
            <Link className="login-text" to="/register">
              Don't have an account? Sign Up here..
            </Link>
          </div>
        </div>
      </div>
    </>
  );
};

export default Login;
