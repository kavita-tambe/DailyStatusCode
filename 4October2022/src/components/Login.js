import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import Avatar from "@mui/material/Avatar";
import Button from "@mui/material/Button";
import CssBaseline from "@mui/material/CssBaseline";
import TextField from "@mui/material/TextField";

import Link from "@mui/material/Link";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import { Alert } from "react-bootstrap";
import { useUserAuth } from "../context/UserAuthContext";
const theme = createTheme();

export default function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");
  const [error, setError] = useState("");
  const { logIn } = useUserAuth();

  const { logOut } = useUserAuth();
  const navigate = useNavigate();
  const { user } = useUserAuth();
  const emailChangeHandler = (e) => {
    setEmail(e.target.value);
  };
  const passwordChangeHandler = (e) => {
    setPassword(e.target.value);
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    const regexEmail = /^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,15}$/;
    if (!email || regexEmail.test(email) === false) {
      setEmailError("Email is not valid");
    } else {
      setEmailError("");
    }
    setError("");
    const regexPassword =
      /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/i;

    if (!password || regexPassword.test(password) === false) {
      setPasswordError("Please enter the valid password");
    } else {
      setPasswordError("");
    }
    try {
      await logIn(email, password);
      navigate("/home");
    } catch (err) {
      setError("Invalid email and password");
    }
  };
  const handleBlurEmail = (e) => {
    if (e.target.value) {
      setEmailError("");
      return true;
    } else {
      setEmailError("Email is required");
    }
  };
  const handleBlurPassword = (e) => {
    if (e.target.value) {
      setPasswordError("");
      return true;
    } else {
      setPasswordError("Password is required");
    }
  };
  const handleFocusEmail = () => {
    setEmailError("");
  };
  const handleFocusPassword = () => {
    setPasswordError("");
  };
  const handleLogout = async () => {
    try {
      await logOut();
      navigate("/");
    } catch (error) {
      console.log(error.message);
    }
  };
  useEffect(() => {
    if (user) {
      handleLogout();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <ThemeProvider theme={theme}>
      <Container component="main" maxWidth="xs">
        <p data-testid="logintext">my form</p>
        <CssBaseline />
        <Box className="login-box">
          <Avatar className="login-logo">
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Login
          </Typography>
          {error && (
            <Alert variant="danger" className="alert">
              {error}
            </Alert>
          )}
          <Box
            component="form"
            data-testid="formone"
            noValidate
            onSubmit={handleSubmit}
            sx={{ mt: 3 }}
          >
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                  onChange={emailChangeHandler}
                  onBlur={handleBlurEmail}
                  onFocus={handleFocusEmail}
                />
                <span className="form-pass">{emailError}</span>
              </Grid>
              <Grid item xs={12}>
                <TextField
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="new-password"
                  onChange={passwordChangeHandler}
                  onBlur={handleBlurPassword}
                  onFocus={handleFocusPassword}
                />
                <span className="form-pass">{passwordError}</span>
              </Grid>
            </Grid>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              className="login-btn"
            >
              Login
            </Button>
            <Grid container justifyContent="flex-end">
              <Grid item>
                <Link href="/register" variant="body2">
                  Don't have an account? Sign Up here..
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </Container>
    </ThemeProvider>
  );
}
