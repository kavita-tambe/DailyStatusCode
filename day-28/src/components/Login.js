import * as React from "react";
import Avatar from "@mui/material/Avatar";
import Button from "@mui/material/Button";
import CssBaseline from "@mui/material/CssBaseline";
import TextField from "@mui/material/TextField";
import FormControlLabel from "@mui/material/FormControlLabel";
import Checkbox from "@mui/material/Checkbox";
import Link from "@mui/material/Link";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import { useNavigate } from "react-router-dom";

const theme = createTheme();

export default function Login() {
    const navigate=useNavigate()
  const handleSubmit = (event) => {
    event.preventDefault();
    
    navigate("/home")
  };

  return (
    <ThemeProvider theme={theme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            padding: "50px",
            boxShadow: "0 3px 10px rgb(0 0 0 / 0.2)",
          }}
        >
          <Avatar sx={{ m: 1, bgcolor: "rgb(255, 38, 37)" }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Login
          </Typography>
          <Box
            component="form"
            noValidate
            onSubmit={handleSubmit}
            sx={{ mt: 3 }}
          >
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="new-password"
                />
              </Grid>
            </Grid>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Login
            </Button>
            <Grid container justifyContent="flex-end">
              <Grid item>
                <Link href="/register" variant="body2">
                  Don't have an account? Sign Up here..
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </Container>
    </ThemeProvider>
  );
}
