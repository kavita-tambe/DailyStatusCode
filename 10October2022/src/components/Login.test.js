import React from "react";
import { shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
// import { render, screen } from "@testing-library/react";
import { useUserAuth } from "../context/UserAuthContext";
import Login from "./Login";
import { BrowserRouter } from "react-router-dom";

import { UserAuthContextProvider } from "../context/UserAuthContext";
describe("render component", () => {
  test("should render component", () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const data = render(
      <BrowserRouter>
        <UserAuthContextProvider>
          <Login />
        </UserAuthContextProvider>
      </BrowserRouter>
    );
    // expect(wrapper.getElements()).toMatchSnapshot();
    expect(data).toBeTruthy();
  });
  test("should render component without error..", () => {
    let componentFun = {
      emailChangeHandler: jest.fn(),
    };
    const wrapper = shallow(
      // <useUserAuth>
      //   <Login {...componentFun} />
      // </useUserAuth>
      <BrowserRouter>
        <UserAuthContextProvider>
          <Login {...componentFun} />
        </UserAuthContextProvider>
      </BrowserRouter>
    );
    expect(wrapper.exists()).toBe(true);
  });
  test("should render component of useState", () => {
    let setEmail = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockReturnValue([true, setEmail]);
    const wrapper = shallow(
      <useUserAuth>
        <Login />
      </useUserAuth>
    );

    wrapper.update();
  });

  test("should render component of useState setPassword", () => {
    let setPassword = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockReturnValue([true, setPassword]);
    const wrapper = shallow(
      <useUserAuth>
        <Login />
      </useUserAuth>
    );

    wrapper.update();
  });
  test("should render component of useState error", () => {
    let setEmailError = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockReturnValue([true, setEmailError]);
    const wrapper = shallow(
      <useUserAuth>
        <Login />
      </useUserAuth>
    );

    wrapper.update();
  });

  test("should render component without error", () => {
    let componentFun = {
      createTheme: jest.fn(),
    };
    const wrapper = shallow(
      <useUserAuth>
        <Login {...componentFun} />
      </useUserAuth>
    );
    expect(wrapper.exists()).toBe(true);
  });

  test("should render component without error...", () => {
    let componentFun = {
      Login: jest.fn(),
    };
    const wrapper = shallow(
      <useUserAuth>
        <Login {...componentFun} />
      </useUserAuth>
    );
    expect(wrapper.exists()).toBe(true);
  });
});
// import React from "react";
// import { shallow } from "enzyme";

// describe("should render component", () => {
//   let wrapper;
//   beforeEach(() => {
//     wrapper = shallow(
//       <useUserAuth>
//         <Login />
//       </useUserAuth>
//     );
//   });

//   let setEmail = jest.fn();
//   const useStateSpy = jest.spyOn(React, "useState");
//   useStateSpy.mockImplementation((init) => [init, setEmail]);

//   let setPassword = jest.fn();
//   const useStateSpypassword = jest.spyOn(React, "useState");
//   useStateSpypassword.mockImplementation((init) => [init, setPassword]);

//   let setEmailError = jest.fn();
//   const useStateSpyemailerr = jest.spyOn(React, "useState");
//   useStateSpyemailerr.mockImplementation((init) => [init, setEmailError]);

//   let setPasswordError = jest.fn();
//   const useStateSpypass = jest.spyOn(React, "useState");
//   useStateSpypass.mockImplementation((init) => [init, setPasswordError]);

//   test("render component without error", () => {
//     let component = wrapper.find("[data-testid='component-list']");
//     expect(component).toBe(true);
//   });
// });
describe("Executes a handler function", () => {
  it("must call the mock method", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <UserAuthContextProvider>
          <Login />
        </UserAuthContextProvider>
      </BrowserRouter>
    );
    const textField = wrapper.find("TextField");
    const instance = wrapper.instance();

    instance.emailChangeHandler = jest.fn(instance.btnClick);

    textField.simulate("email");
    expect(instance.emailChangeHandler).toHaveBeenCalled();
  });
});
