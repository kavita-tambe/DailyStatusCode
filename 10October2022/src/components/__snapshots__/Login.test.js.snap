// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component should render component 1`] = `
Array [
  <useUserAuth>
    <Login />
  </useUserAuth>,
]
`;
